### PR TITLE
[ios] Fix links in documentation

### PR DESCRIPTION
**Pull request by captainbarbosa**
Fixes two stale links within the iOS documentation.

No changelog entry required for this PR.

### CODE REVIEW

**Comment by friedbunny** [comment]
You might also update: https://github.com/mapbox/mapbox-gl-native/blob/d5659aa6647f1fc77159567bd22029a2dc9cd7a3/platform/ios/src/MGLSDKUpdateChecker.mm#L23

File: platform/ios/docs/guides/Info.plist Keys.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ The default value is `https://api.mapbox.com`.
 
 ## MGLMapboxMetricsEnabledSettingShownInApp
 
-If you have implemented custom opt-out of Mapbox Telemetry within the user interface of your app, use this key to disable the built-in check for opt-out support. See [this guide](https://www.mapbox.com/ios-sdk/#telemetry_opt_out) for more details.
+If you have implemented custom opt-out of Mapbox Telemetry within the user interface of your app, use this key to disable the built-in check for opt-out support. See [this guide](https://www.mapbox.com/help/how-mapbox-works/attribution/#mapbox-maps-sdk-for-ios) for more details.
```
Comment (1ec5) [inline platform/ios/docs/guides/Info.plist Keys.md]: This page has moved [yet again](https://docs.mapbox.com/help/how-mapbox-works/attribution/#mapbox-maps-sdk-for-ios).

**Comment by friedbunny** [comment]
Let’s get this into `release-java`.

**Comment by captainbarbosa** [comment]
I've also updated all `mapbox.com/help` URLs to point to the new `docs.mapbox.com` format in this PR.

**Comment by riastrad** [comment]
Quick fly-by to note that we'll also need to update all of the 'related example' references I added in https://github.com/mapbox/mapbox-gl-native/pull/13236 to reflect the new sub-domain url structure. I'll open up a separate issue to track that effort.

**Pull request merged**